http://bugs.gentoo.org/447182
http://bugzilla.xfce.org/show_bug.cgi?id=9615

From 48e1016394fa6bdb1d708823a23a12641482a561 Mon Sep 17 00:00:00 2001
From: Carlos Silva <r3pek@r3pek.org>
Date: Fri, 14 Dec 2012 01:14:36 +0000
Subject: Fix icons not probably resizing when requested.

Looks like gdk_pixbuf_new_from_file_at_scale and gdk_pixbuf_new_from_file_at_size really don't scale the image leaving the image loaded with its original size. So, just load the file without any scale and let the scale be done later on the function.

Signed-off-by: Carlos Silva <r3pek@r3pek.org>
---
diff --git a/libxfce4panel/xfce-panel-convenience.c b/libxfce4panel/xfce-panel-convenience.c
index 19c51bb..a3f3be1 100644
--- a/libxfce4panel/xfce-panel-convenience.c
+++ b/libxfce4panel/xfce-panel-convenience.c
@@ -170,15 +170,13 @@ xfce_panel_pixbuf_from_source_at_size (const gchar  *source,
 
   if (G_UNLIKELY (g_path_is_absolute (source)))
     {
-      pixbuf = gdk_pixbuf_new_from_file_at_scale (source, dest_width, dest_height, TRUE, &error);
+      pixbuf = gdk_pixbuf_new_from_file (source, &error);
       if (G_UNLIKELY (pixbuf == NULL))
         {
           g_message ("Failed to load image \"%s\": %s",
                      source, error->message);
           g_error_free (error);
         }
-
-      return pixbuf;
     }
   else
     {
--
cgit v0.9.0.3
